### DISCUSSION TITLE
weird subclass / subproperty construction in rdfs output of OntoMe isn't usable by reasoners

### DISCUSSION THREAD

**Discussion by Habennin** [geovistory/.github#16] [Bugs]
So this is an OntoMe issue, not geovistory, as far as I understand it.

But here is the problem: when generating RDF from an OntoMe project, one hopes it will work with CIDOC CRM based on declaring the subclass and subproperty relations.

But in the tools I use (3M and Arches in this case) the reasoners do not handle the way that OntoMe builds the subclass and subproperty relations.

This is an example of the structure that OntoMe automatically builds for a subclass relation:

 <rdfs:subClassOf>
      <rdfs:Class rdf:about="http://www.cidoc-crm.org/cidoc-crm/E2_Temporal_Entity"/>
 </rdfs:subClassOf>

This is an example of the structure that OntoMe automatically builds for a subproperty relation:

 <rdfs:subPropertyOf>
      <rdf:Property rdf:about="https://takin.solutions/ontologies/crmaaa/ZP1_has_intentional_subject"/>
</rdfs:subPropertyOf>

Neither of these create a connection to their respective super classes or super properties meaning that you can't use it as an ontology!

The syntax that does work (and is used in all cidoc crm serializations at least) is this:

<rdfs:subClassOf rdf:resource="http://www.cidoc-crm.org/cidoc-crm/E2_Temporal_Entity"/>

 <rdfs:subPropertyOf rdf:resource="https://takin.solutions/ontologies/crmaaa/ZP2_ascribes_intentional_target" />

I am not asserting that the first syntax is wrong. I may be ignorant and that is also a valid way of making the same statement. I am saying that it is incompatible with the way that the official cidoc crm serialization works. So the extensions that one makes with OntoMe are functionally incompatible with CIDOC CRM in their automatic serialization.

To me this seems not ideal. 

What are other people doing about this?

Is it possible to change the syntax so that it is compatible ( i now have to do this by hand, meaning that OntoMe is not doing the job of automatically making the rdfs).

**Comment by joschne** [comment]
Hi @Habennin 

Indeed, the issue you describe is also affecting other people. In WissKI, for example, inheritance of properties is not working with the OWL DL export of OntoME. Are you having the issue with the OWL DL or the RDFS export of OntoME?

In case you use OLW DL, is it correct that by following the pattern of the following owl, it should work with 3M/Arches?
https://gitlab.isl.ics.forth.gr/cidoc-crm/cidoc_crm_rdf/-/blob/v7.1.2_owl_preparation/7.1.2/owl/CIDOC_CRM_v7.1.2.owl

**Comment by atterebf** [comment]
The OWL serializaion available today in OntoME comes from the [Erlangen CRM specification from the years 2018-2019](https://github.com/erlangen-crm/ecrm). This was by then the only OWL serialisation existing and we used that one in OntoME.

According to the [OWL 2 specification](https://www.w3.org/TR/owl2-primer/#OWL_Syntaxes), these are the to wished serialisations:

Functional-Style Syntax
 SubClassOf( :Mother :Woman )

RDF/XML Syntax
 <owl:Class rdf:about="Mother">
   <rdfs:subClassOf rdf:resource="Woman"/>
 </owl:Class>

Functional-Style Syntax
 SubObjectPropertyOf( :hasWife :hasSpouse )

RDF/XML Syntax
 <owl:ObjectProperty rdf:about="hasWife">
   <rdfs:subPropertyOf rdf:resource="hasSpouse"/>
 </owl:ObjectProperty>

Generally speaking I'd suggest that the Geovistory data team reconsiders the whole serialisation of OntME classes and properties accoding to the standard specification of OWL 2, then compares and discusses the nowadays existing serialisation of CRM, and then decides what to implement.

**Comment by Habennin** [comment]
Dear both,

I guess we also need to distinguish between owl serialization and rdfs serialization. OntoMe also offers both these options as it currently stands. This seems good practice since not everyone uses the same serialization. See here:

<img width="1491" alt="Screenshot 2024-02-06 at 12 27 11 PM" src="https://github.com/geovistory/.github/assets/12068255/8fe19df0-4c56-46a7-8d92-ef271717696e">

I do not know about the OWL-2 serialization but I take it of course that @atterebf  is right in what he argues.

Re RDFS, I do believe that the present implementation of RDFS in the official CIDOC CRM rdfs files is correct and follows good practice. If I am wrong about that it would be good to know and it should be raised to the CIDOC CRM community. This is where that serialization can be acquired:

<img width="1491" alt="Screenshot 2024-02-06 at 12 15 23 PM" src="https://github.com/geovistory/.github/assets/12068255/1ba16109-cbe5-439f-93a7-0770578ffa24">

here is the direct link: 

https://cidoc-crm.org/rdfs/7.1.2/CIDOC_CRM_v7.1.2.rdfs

And this is what the structure looks like there:

 <rdfs:subClassOf rdf:resource="E1_CRM_Entity" />

It should be noted that it is those RDFS serializations (from the official site/organization) that people use to implement CIDOC CRM in RDFS (and should continue to use) so if what OntoMe produces is a serialization that is incompatible with that then it does not support compatibility with CIDOC CRM (at that very useful functional level of being able to reallly use it at least). This is a problem insofar as that is a goal, but since I take it that it is indeed a goal, in principle, I would bear that in mind. I know when I direct people to OntoMe as a software this is one of the features I stress that it supports and why I encourage them to use it. It provides stable, community guided ways of generating CRM extensions that you can then use with CIDOC CRM's official serializations.  Also, if 'official' extensions of CRM are to be managed in OntoMe this would be a condition for submitting them to the community that there RDFS is conformant with all the other serializations. I only mention this last as I can imagine that this is not a high goal given the extremely frustrating resistance to adoption met there.

In this and other tickets I have proposed a number of changes that would bring the existing serialization into line with the CIDOC CRM RDFS serialization practice (which again I think actually does follow the rules in this case!). To my estimation they are actually not particularly big changes to someone who has already written the serialization code. A few more if thens, something appearing here instead of there. etc. 

I would be happy to consolidate all those remarks but they are just basically about OntoMe being able to generate usable rdfs for the CIDOC community following the standard.

Thanks for looking into the matter.

Sincerely,

George

**Comment by atterebf** [comment]
Dear George, thank you for raising all these issues.
I thougth the discussion was about the OWL serialization. We will of course check also the RDFS one and of course consider all the relevant points you make.
Best

**Comment by Habennin** [comment]
Dear Francesco,

Thank you for your comment. I do not feel very confident with the OWL serialization to be sure and I defer to your knowledge. 

I do feel like I have a strong practical knowledge of how the RDFS serialization typically works and my remarks only refer to this scope of the question. I also accept I could have missed a step in the evolution of the standard, but to my knowledge the way it is implemented in the present CRM base and extension is still proper. 

I remain a strong advocate for and fan of the vision of OntoMe! I hope we can continue the discussion and see if there is room in the work on OntoMe to continue to improve the functionality of the tool for different audiences and their serialization needs.

Cheers,
George

**Reply by atterebf** [reply]
As far as I can see in Protégé, the RDFS serialization works correctly. Do you see a problem in 3M and Arches ?

**Reply by Habennin** [reply]
I do. It doesn't recognize the subclass / superclass relation, so you can't use the ontology features of the ontology.

**Reply by Habennin** [reply]
after manually rewriting them to the form that CIDOC CRM uses, they work. This of course is time intensive and causes human made errors that OntoMe could facillitate not happening.

**Reply by Habennin** [reply]
I could screen record it if that helps.  3M uses exist queries in order to parse the rdf/xml and render the correct syntactic relations between classses and properties. It seems it doesn't to recognize this version of the representation of subclass/superclass. I do not know what Arches uses to parse the rdf, I would have to look into it.

I should note that I may also be introducing errors into the RDFS because after OntoMe produces it, I need to hand edit it to put in various basic features that aren't there (documented in the other issues),like inverse properties. I run the file through rdf validation and it works, but there is always that chance I suppose.

**Reply by Habennin** [reply]
What makes me think I'm not crazy is that a) the machine doesn't work until I make this change (to the style adopted in all other rdfs that I use) and b) it seems to be affecting not just me.

**Reply by atterebf** [reply]
What I ment is: did you try today? with the serialisation of today? :-)

**Reply by Habennin** [reply]
oh I'm sorry has it changed already? I will look!

**Reply by Habennin** [reply]
@atterebf I wanted to confirm that the change introduced into the RDFS output generator has fixed the issue that started this issue. I tested with 3M and it recognizes the position of a subclass to the CRM hierarchy and delivers the proper set of inherited relations.

Thanks a lot to everyone who had a look into it

![Uploading Screenshot 2024-02-15 at 4.01.57 PM.png…]()

**Comment by joschne** [comment]
> In this and other tickets I have proposed a number of changes that would bring the existing serialization into line with the CIDOC CRM RDFS serialization practice (which again I think actually does follow the rules in this case!). To my estimation they are actually not particularly big changes to someone who has already written the serialization code. A few more if thens, something appearing here instead of there. etc.

Thank you @Habennin for these issues. You created them [in this repo](https://github.com/Semantic-Data-for-Humanities/OntoME/issues?q=is%3Aissue+is%3Aclosed), and they have been transferred here to [discussions](https://github.com/orgs/geovistory/discussions). 
(The discussions will become the one channel for discussions about the entire Geovistory infrastructure, including OntoME.)

In order to group the (related) discussions I created a label `RDFS and OWL export`, see [this list](https://github.com/orgs/geovistory/discussions?discussions_q=is%3Aopen+label%3A%22RDFS+and+OWL+export%22).

As next steps I propose that we
- create two new OntoME issues, one for RDFS, one for OWL export
- link the issues in the corresponding discussions 
- close the discussions in favor of the new issue
- write the issue spec, by summarizing the issues reported by @Habennin and by critically reviewing the overall exported XML's within the data team
- let @Habennin review the RDFS issue spec before implementation (if ok for him)

Can you take care of this @valamercery ?

**Reply by Habennin** [reply]
@joschne I'd be happy to participate
